src/register.jl

```diff
@@ -39,18 +39,18 @@ export ArrayReg,
     oneto
 
 """
-    ArrayReg{B, T, MT <: AbstractMatrix{T}} <: AbstractRegister{B, T}
+    ArrayReg{B, T, MT <: AbstractMatrix{T}} <: AbstractRegister{B}
 
 Simulated full amplitude register type, it uses an array to represent
 corresponding one or a batch of quantum states. `B` is the batch size, `T`
 is the numerical type for each amplitude, it is `ComplexF64` by default.
 """
-mutable struct ArrayReg{B, T, MT <: AbstractMatrix{T}} <: AbstractRegister{B, T}
+mutable struct ArrayReg{B, T, MT <: AbstractMatrix{T}} <: AbstractRegister{B}
     state::MT
 end
 
-const AdjointArrayReg{B, T, MT} = AdjointRegister{B, T, ArrayReg{B, T, MT}}
-const ArrayRegOrAdjointArrayReg{B, T, MT} = Union{ArrayReg{B, T, MT}, AdjointRegister{B, T, ArrayReg{B, T, MT}}}
+const AdjointArrayReg{B, T, MT} = AdjointRegister{B, ArrayReg{B, T, MT}}
```

Comment:
this was not fixed in last PR?

Reply:
I don't think so.No worries,this Is not a big deal, just a redundant type parameter in adjoint type.